circuits/cpp/barretenberg/cpp/src/barretenberg/bb/get_crs.hpp

```diff
@@ -62,8 +62,15 @@ inline std::vector<barretenberg::g1::affine_element> get_g1_data(const std::file
             vinfo("using cached crs at: ", path);
             auto data = read_file(path / "g1.dat");
             auto points = std::vector<barretenberg::g1::affine_element>(num_points);
+
+            auto size_of_points_in_bytes = num_points * 64;
+            if (data.size() < size_of_points_in_bytes) {
+                vinfo("data is smaller than expected!", data.size(), size_of_points_in_bytes);
+            }
+            size_t actual_buffer_size = std::min(data.size(), size_of_points_in_bytes);
```

Comment:
In the case where size > num_points which should imply that data.size() > num_points * 64, this will have the same behaviour as before